8.0 pxc 3444

nitendra-bhosle opened the pull request:
PXC-3444: Node crashes on TOI DML rollback
    
https://jira.percona.com/browse/PXC-3444

Analysis :
Node crashes while DML is executed in TOI mode,
and getting rollback.
This behavior is specific for PXC only where DML can be replicated in TOI mode. 
This is used by pt-table-checksum tool which injects 99997 version 
into the insert.. select query.

Fix:
Handled m_toi transaction state on rollback,
by aborting transaction while DML is executed in TOI mode.

kamil-holubicki reviewed (changes requested):
Hi, the change looks fine, but please write a detailed explanation in the commit message (and maybe something inside the code as well) why we need it.
This behavior is specific for PXC only where DML can be replicated in TOI mode. This is used by pt-table-checksum tool which injects 99997 version into the insert.. select query

kamil-holubicki commented:
Please merge just approved https://github.com/percona/wsrep-lib/pull/8, update wsrep repo poiner, squash commits, update commit message as requested.

dutow reviewed (changes requested):
As we discussed on slack, please do not merge this as is into the 8.0 branch, because the wsrep-lib submodule also contains changes for the 8.0.25 merge.

Either target the PR to the release-8.0.25 branch, or wait until we merge back the release branch.

nitendra-bhosle commented:
> As we discussed on slack, please do not merge this as is into the 8.0 branch, because the wsrep-lib submodule also contains changes for the 8.0.25 merge.
> 
> Either target the PR to the release-8.0.25 branch, or wait until we merge back the release branch.

Yeah , sure . I will revert these changes   and wait for release branch . This bug will be fixed only if we get changes from wsrep-lib library .

kamil-holubicki commented:
I don't know which wsrep-lib hash goes into 8.0.25, but if the change related to PXC-3444 is included, I would merge this PR to 8.0.25 release branch as well (here we have only tests, fix is in wsrep-lib).
On the flip side, if 8.0.25 wsrep-lib does not contain fix for PXC-3444, as Zsolt said, we need to wait with merging this PR till 8.0.25 release branch is merged back to 8.0.

Other than that, all looks good, please squash before merging.

kamil-holubicki reviewed (approved):
LGTM

nitendra-bhosle commented:
Jenkins : https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/370/

nitendra-bhosle commented:
New Jenkins : https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/392/

Commit:
PXC-3444: Node crashes on TOI DML rollback

https://jira.percona.com/browse/PXC-3444

Analysis :
Node crashes while DML is executed in TOI mode,
and getting rollback.
This behavior is specific for PXC only where DML can be replicated in TOI mode.
This is used by pt-table-checksum tool which injects 99997 version
into the insert.. select query.

Fix:
Handled m_toi transaction state on rollback,
by aborting transaction while DML is executed in TOI mode.
Added test case.
